src/adagrams.js

```diff
@@ -1,15 +1,157 @@
+
+const score_chart = {
+  'A': 1,
+  'B': 3,
+  'C': 3,
+  'D': 2,
+  'E': 1,
+  'F': 4,
+  'G': 2,
+  'H': 4,
+  'I': 1,
+  'J': 8,
+  'K': 5,
+  'L': 1,
+  'M': 3,
+  'N': 1,
+  'O': 1,
+  'P': 3,
+  'Q': 10,
+  'R': 1,
+  'S': 1,
+  'T': 1,
+  'U': 1,
+  'V': 4,
+  'W': 4,
+  'X': 8,
+  'Y': 4,
+  'Z': 10
+}
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let LETTER_POOL = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+  };
+  // create a string that holds all the avaliable letters - including probability
+  let playerHand = [];
+  let alphabet = "";
+  for (const [key, value] of Object.entries(LETTER_POOL)) {
+    alphabet += key.repeat(value);
+  }
+
+  // create 10 letters for the player hand
+  // subtract the letter from avaliable letters
+  while (playerHand.length < 10) {
+    let randomCharacter = alphabet[Math.floor(Math.random() * alphabet.length)];
+
+    if (LETTER_POOL[randomCharacter] === 0) {
+      continue;
+    } else {
+      playerHand.push(randomCharacter);
+      LETTER_POOL[randomCharacter] -= 1;
+      alphabet.replace(randomCharacter, '');
```

Comment:
love this idea! One thing I would suggest is turning `alphabet` into an array instead of a string. That way when we change the index position's value to an empty string, we aren't creating a whole new copy of the data structure the way `replace` is doing.

Each time we call `replace` it has to make a new copy of the string.